ktor-client/ktor-client-mock/common/src/io/ktor/client/engine/mock/MockUtils.kt

```diff
@@ -12,14 +12,22 @@ import io.ktor.util.date.*
 import io.ktor.utils.io.*
 import io.ktor.utils.io.charsets.*
 import io.ktor.utils.io.core.*
+import kotlinx.coroutines.*
 
+@OptIn(DelicateCoroutinesApi::class)
 @Suppress("KDocMissingDocumentation")
 public suspend fun OutgoingContent.toByteArray(): ByteArray = when (this) {
     is OutgoingContent.ByteArrayContent -> bytes()
     is OutgoingContent.ReadChannelContent -> readFrom().toByteArray()
     is OutgoingContent.WriteChannelContent -> {
-        ByteChannel().also { writeTo(it) }.toByteArray()
+        val channel = ByteChannel()
+        GlobalScope.launch(Dispatchers.Unconfined) {
+            writeTo(channel)
+            channel.close()
+        }
+        channel.toByteArray()
     }
```

Comment:
I know this has already been merged, but shouldn't structured concurrency be used here (and in `toByteReadPacket`) instead of `GlobalScope`:
```kotlin
is OutgoingContent.WriteChannelContent -> coroutineScope {
    val channel = ByteChannel()
    launch(start = UNDISPATCHED) { /* ... */ }
    channel.toByteArray()
}
```